### PR TITLE
refactor: Deprecate ddof parameter for correlation coefficient

### PR DESCRIPTION
**Pull request by flowlight0**
Fix https://github.com/pola-rs/polars/issues/19420

### CODE REVIEW

File: py-polars/polars/functions/lazy.py [hunk 1]
```diff
@@ -794,9 +794,10 @@ def corr(
     b
         Column name or Expression.
     ddof
-        "Delta Degrees of Freedom": the divisor used in the calculation is N - ddof,
-        where N represents the number of elements.
-        By default ddof is 1.
+        Has no effect, do not use.
+
+        .. deprecated:: 1.17.0
+
```
Comment (flowlight0) [inline py-polars/polars/functions/lazy.py:800]: I'm not confident about choice the version written here. I just incremented from the current release version 1.16.0.

File: crates/polars-plan/src/dsl/functions/correlation.rs [hunk 2]
```diff
@@ -20,15 +20,11 @@ pub fn cov(a: Expr, b: Expr, ddof: u8) -> Expr {
...
-/// * ddof
-///     Delta degrees of freedom
-pub fn pearson_corr(a: Expr, b: Expr, ddof: u8) -> Expr {
+pub fn pearson_corr(a: Expr, b: Expr) -> Expr {
     let input = vec![a, b];
     let function = FunctionExpr::Correlation {
         method: CorrelationMethod::Pearson,
-        ddof,
+        ddof: 0u8,
     };
```
[the remark this answers is not in the document]
Comment (ritchie46) [reply crates/polars-plan/src/dsl/functions/correlation.rs:28]: Yes, we can remove `ddof` there I think.

[the remark this answers is not in the document]
Comment (flowlight0) [reply crates/polars-plan/src/dsl/functions/correlation.rs]: Thanks for your comment! I moved ddof parameter in `FunctionExpr::Correlation` into `CorrellationMethod::Covariance` (https://github.com/pola-rs/polars/pull/20197/commits/f00ff2bb479de740a369defd7eca34638a9aa9af) .

**Comment by ritchie46** [comment]
Thank you @flowlight0 :+1:

**Comment by alexander-beedie** [comment]
初めてのPR、ありがとうございます！

**Pull request merged**